homeassistant/components/advantage_air/cover.py

```diff
@@ -0,0 +1,144 @@
+        self.zone_key = zone_key
+
+    @property
+    def _zone(self):
+        return self.coordinator.data["aircons"][self.ac_key]["zones"][self.zone_key]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f'{self._zone["name"]}'
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-{self.zone_key}'
+
+    @property
+    def device_class(self):
+        """Return the device class of the vent."""
+        return DEVICE_CLASS_DAMPER
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+
+    @property
+    def is_closed(self):
+        """Return if vent is fully closed."""
+        return self._zone["state"] == ADVANTAGE_AIR_STATE_CLOSE
+
+    @property
+    def current_cover_position(self):
+        """Return vents current position as a percentage."""
+        if self._zone["state"] == ADVANTAGE_AIR_STATE_OPEN:
+            return self._zone["value"]
+        return 0
+
+    @property
+    def icon(self):
+        """Return vent icon."""
+        if self._zone["state"] == ADVANTAGE_AIR_STATE_OPEN:
+            return "mdi:fan"
+        return "mdi:fan-off"
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return {
+            "identifiers": {(DOMAIN, self.coordinator.data["system"]["rid"])},
+            "name": self.coordinator.data["system"]["name"],
+            "manufacturer": "Advantage Air",
+            "model": self.coordinator.data["system"]["sysType"],
+            "sw_version": self.coordinator.data["system"]["myAppRev"],
+        }
+
+    async def async_open_cover(self, **kwargs):
+        """Fully open zone vent."""
+        await self.async_change(
+            {
+                self.ac_key: {
+                    "zones": {
+                        self.zone_key: {"state": ADVANTAGE_AIR_STATE_OPEN, "value": 100}
+                    }
+                }
+            }
+        )
+
+    async def async_close_cover(self, **kwargs):
+        """Fully close zone vent."""
+        await self.async_change(
+            {
+                self.ac_key: {
+                    "zones": {self.zone_key: {"state": ADVANTAGE_AIR_STATE_CLOSE}}
+                }
+            }
+        )
+
+    async def async_set_cover_position(self, **kwargs):
+        """Change vent position."""
+        position = round(kwargs.get(ATTR_POSITION) / 5) * 5
+        if position == 0:
+            await self.async_change(
+                {
+                    self.ac_key: {
+                        "zones": {self.zone_key: {"state": ADVANTAGE_AIR_STATE_CLOSE}}
+                    }
+                }
+            )
+        else:
+            await self.async_change(
+                {
+                    self.ac_key: {
+                        "zones": {
```

Comment:
Again depending prefrence, but you could set `ZONES = "zones"` locally (orin const and then import them). again same with states/states, but that's preference and probability of change related.

Reply:
I am unsure what impact that could have on the API itself, as it can support multiple changes at once, but it's documentation says to send only the values that need to be changed.

I worry sending the entire zone may cause the hardware to try change every value, some of which are read-only.

Reply:
No I meant replacing the x times "zones" with a const

So more like (ATTR_STATE is a homeassistant core const):
```
            await self.async_change(
                {
                    self.ac_key: {
                       ZONES: {
                             self.zone_key: {ATTR_STATE: ADVANTAGE_AIR_STATE_OPEN,
```

Reply:
Oh right, yeah that's another good idea. Is this a memory optimisation for Python, having to store the string only once?

I won't use STATE_ATTR as Martin has previously told me not to use HA constants for my APIs strings. But totally could add zone, AC, state, system etc as constants in const.py

Reply:
Ah right, indeed - not allowed to do it there, but it's not optimisation, rather if something changes you'll only have to change the const and not a lot of occurances.

Reply:
I think I'll have to address this in a wider code quality update as its going to need to touch all platforms and tests.